metricflow/model/transformations/convert_median.py

```diff
@@ -0,0 +1,22 @@
+from metricflow.aggregation_properties import AggregationType
+from metricflow.model.objects.elements.measure import MeasureAggregationParameters
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.transformations.transform_rule import ModelTransformRule
+
+MEDIAN_PERCENTILE = 0.5
+
+
+class ConvertMedianToPercentileRule(ModelTransformRule):
+    """Converts any MEDIAN measures to percentile equivalent."""
+
+    @staticmethod
+    def transform_model(model: UserConfiguredModel) -> UserConfiguredModel:  # noqa: D
+        for data_source in model.data_sources:
+            for measure in data_source.measures:
+                if measure.agg == AggregationType.MEDIAN:
+                    measure.agg = AggregationType.PERCENTILE
+                    if not measure.agg_params:
+                        measure.agg_params = MeasureAggregationParameters()
+                    measure.agg_params.percentile = MEDIAN_PERCENTILE
+                    measure.agg_params.disc = False
```

Comment:
Let's assert that these are not set to different values. Like if someone says they want a median with percentile=0.25 they should get an error telling them that this is conflicting and they should remove the percentile parameter.

We can allow them to code the correct values in the model, that's fine.